### PR TITLE
Make games new

### PR DESCRIPTION
**Pull request by freerangeandy**
Added React form on games index page for submitting new games
Added post fetch request to games index component
Created endpoint tests for POST#create route

### CODE REVIEW

File: app/controllers/api/v1/games_controller.rb [hunk 1]
```diff
@@ -6,4 +6,21 @@ def index
...
     render json: Game.find(params[:id])
   end
+
+  def create
+    game = Game.new(game_params)
+
+    if game.save
+      render json: game
+    else
+      render json: {error: "Game did not save"}
```
Comment (safang99) [inline app/controllers/api/v1/games_controller.rb:16]: Lets use `game.errors` to give descriptive feedback and make use of  the validations

File: spec/controllers/api/v1/games_controller_spec.rb [hunk 2]
```diff
@@ -49,4 +49,68 @@
...
+
+      post_json = {
+        game: {
+          title: "Basset Hound Shakes Off",
+        }
+      }
+
+      prev_count = Game.count
+
+
```
Comment (safang99) [inline spec/controllers/api/v1/games_controller_spec.rb:64]: lets remove on of the duplicate blank lines

File: spec/controllers/api/v1/games_controller_spec.rb [hunk 3]
```diff
@@ -49,4 +49,68 @@
...
+        game: {
+          title: ""
+        }
+      }
+
+      prev_count = Game.count
+      post(:create, params: post_json)
+      expect(Game.count).to eq(prev_count)
+
+      expect(response.status).to eq 200
```
Comment (safang99) [inline spec/controllers/api/v1/games_controller_spec.rb:113]: This test isnt actually checking for errors.  After changing the errors returned on the controller, lets test that the errors being returned are what you expect when required fields are not filled in.

**Comment by BenjaminFairbank** [comment]
Resolved issues above

**Pull request merged**